src/SmartComponents/Advisories/Advisories.test.js

```diff
@@ -55,152 +59,66 @@ const mockState = { ...storeListDefaults,
 };
 
 const initStore = (state) => {
-    const customMiddleWare = () => next => action => {
-        useSelector.mockImplementation(callback => {
-            return callback({  AdvisoryListStore: state });
-        });
-        next(action);
-    };
-
-    const mockStore = configureStore([customMiddleWare]);
+    const mockStore = configureStore([]);
     return mockStore({  AdvisoryListStore: state });
 };
 
-let wrapper;
 let store = initStore(mockState);
 
 beforeEach(() => {
-    store.clearActions();
-    useSelector.mockImplementation(callback => {
-        return callback({ AdvisoryListStore: mockState });
-    });
-    wrapper = mount(<Provider store={store}>
-        <Router><Advisories/></Router>
-    </Provider>);
-});
-
-afterEach(() => {
-    useSelector.mockClear();
+    render(<ComponentWithContext renderOptions={{ store }}>
+        <Advisories/>
+    </ComponentWithContext>);
 });
 
 describe('Advisories.js', () => {
+    testExport(exportAdvisoriesCSV, exportAdvisoriesJSON, 'advisories');
```

Comment:
nit-pick: even though the test is shared between two different test files, I'd better have it separately duplicated in both of them. Reading this/traveling to another file module/understanding props just increases the cognitive load and makes the code less readable. The benefits of having DRY code are also not significant. IMHO

Reply:
I would argue that having duplicate tests for the same feature would require me to to each test file and make the same changes. That is more time-consuming and cumbersome than cognitive load :)